Examples/SnapinsSDKKotlinExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/activities/MainActivity.java

```diff
@@ -0,0 +1,269 @@
+package com.salesforce.snapinssdkexample.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.salesforce.android.chat.core.ChatConfiguration;
+import com.salesforce.android.chat.core.ChatCore;
+import com.salesforce.android.chat.core.model.AvailabilityState;
+import com.salesforce.android.knowledge.ui.KnowledgeUI;
+import com.salesforce.android.knowledge.ui.KnowledgeUIClient;
+import com.salesforce.android.service.common.analytics.ServiceAnalytics;
+import com.salesforce.android.service.common.analytics.ServiceAnalyticsListener;
+import com.salesforce.android.service.common.utilities.control.Async;
+import com.salesforce.android.sos.api.SosAvailability;
+import com.salesforce.androidsdk.app.SalesforceSDKManager;
+import com.salesforce.androidsdk.rest.ClientManager;
+import com.salesforce.androidsdk.rest.RestClient;
+import com.salesforce.snapinssdkexample.R;
+import com.salesforce.snapinssdkexample.SupportHomeViewAddition;
+import com.salesforce.snapinssdkexample.activities.settings.CaseSettingsActivity;
+import com.salesforce.snapinssdkexample.activities.settings.ChatSettingsActivity;
+import com.salesforce.snapinssdkexample.activities.settings.KnowledgeSettingsActivity;
+import com.salesforce.snapinssdkexample.activities.settings.SosSettingsActivity;
+import com.salesforce.snapinssdkexample.utils.ServiceSDKUtils;
+
+import java.util.Map;
+
+/**
+ * Main test activity supporting basic functionality:
+ * <ul>
+ * <li>Launching Knowledge UI {@link SupportHomeViewAddition}</li>
+ * <li>Authentication (Logging in or out)</li>
+ * </ul>
+ */
+public class MainActivity extends AppCompatActivity implements SosAvailability.Listener {
+    private KnowledgeUI mKnowledgeUI = null;
```

Comment:
Nit: No need to initialize these as null, it reads a little weird (in my opinion).

Reply:
done